src/main/java/org/sagebionetworks/web/client/SynapseJavascriptClient.java

```diff
@@ -1847,10 +1849,53 @@ public void getAccessApprovalNotifications(String requirementId, List<String> re
 		String url = getRepoServiceUrl() + ACCESS_APPROVAL + "/notifications";
 		doPost(url, request, OBJECT_TYPE.AccessApprovalNotificationResponse, true, cb);
 	}
-	
+
 	public void getDefaultUploadDestination(String parentEntityId, AsyncCallback<UploadDestination> cb) {
 		String url = getFileServiceUrl() + ENTITY + "/" + parentEntityId + "/uploadDestination";
 		doGet(url, OBJECT_TYPE.UploadDestination, cb);
 	}
+
+	public void getSchemaBinding(String entityId, AsyncCallback<JsonSchemaObjectBinding> cb) {
+		String url = getRepoServiceUrl() + ENTITY + "/" + entityId + "/schema/binding";
+		doGet(url, OBJECT_TYPE.JsonSchemaObjectBinding, cb);
+	}
+
+	public void getSchemaValidationResults(String entityId, AsyncCallback<ValidationResults> cb) {
+		String url = getRepoServiceUrl() + ENTITY + "/" + entityId + "/schema/validation";
+		doGet(url, OBJECT_TYPE.ValidationResults, cb);
+	}
+
+	private void getSchemaValidationResultsWithMatchingEtag(String entityId, String expectedEtag, int delayMs, AsyncCallback<ValidationResults> cb) {
+		gwt.scheduleExecution(
+				() ->
+						getSchemaValidationResults(entityId, new AsyncCallback<ValidationResults>() {
+							@Override
+							public void onSuccess(ValidationResults result) {
+								// If the result eTag and the entity eTag are different, then schema conformance has not been determined yet
+								// re-fetch with back-off until it is consistent
+								if (result.getObjectEtag().equals(expectedEtag)) {
+									cb.onSuccess(result);
+								} else {
+									getSchemaValidationResultsWithMatchingEtag(entityId, expectedEtag, delayMs * 2, cb);
+								}
+							}
+							@Override
+							public void onFailure(Throwable caught) {
+								cb.onFailure(caught);
+							}
+						}),
+				delayMs);
+	}
+
+	/**
+	 * Retries (with exponential backoff) on mismatched eTag or 404
+	 * @param entityId
+	 * @param expectedEtag
+	 * @param cb
+	 */
+	public void getSchemaValidationResultsWithMatchingEtag(String entityId, String expectedEtag, AsyncCallback<ValidationResults> cb) {
+		getSchemaValidationResultsWithMatchingEtag(entityId, expectedEtag, INITIAL_RETRY_REQUEST_DELAY_MS, cb);
+	}
```

Comment:
Schema validation is eventually consistent, where the eTag of the object will match the eTag in the validation results if it's up-to-date. So we want to retry (w/ backoff) on a 404 (assuming there is a schema binding) or eTag mismatch.